### PR TITLE
A variety of small fixes

### PR DESCRIPTION
**Pull request by mikehearn**
* RPC docs improvements (mostly moving stuff around)
* Make RPCClient.RPCConnection a top level public interface
* Make package descriptions appear in Kotlin API docs
* Flesh out docs on ServiceHub
* Upgrade the Gradle version check plugin to make it work again
* Add an IntelliJ scope that covers the public API, useful for spotting missing kdocs.
* Suppress Any extension functions that are internal.

### CODE REVIEW

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt [hunk 1]
```diff
@@ -10,31 +10,62 @@ import net.corda.nodeapi.ConnectionDirection
...
-/** @see RPCClient.RPCConnection */
-class CordaRPCConnection internal constructor(
-        connection: RPCClient.RPCConnection<CordaRPCOps>
-) : RPCClient.RPCConnection<CordaRPCOps> by connection
+/**
+ * This class is essentially just a wrapper for an RPCConnection<CordaRPCOps> and can be treated identically.
+ *
+ * @see RPCConnection
+ */
+class CordaRPCConnection internal constructor(connection: RPCConnection<CordaRPCOps>) : RPCConnection<CordaRPCOps> by connection
```
Comment (shamsasari) [inline client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: As I found out internal constructors are completely visible in Java.

File: core/src/main/kotlin/net/corda/core/node/NodeInfo.kt [hunk 2]
```diff
@@ -6,21 +6,34 @@ import net.corda.core.serialization.CordaSerializable
 import net.corda.core.utilities.NetworkHostAndPort
 
 /**
- * Info about a network node that acts on behalf of some form of contract party.
- * @param legalIdentitiesAndCerts is a non-empty list, where the first identity is assumed to be the default identity of the node.
+ * Information about a network node that acts on behalf of some party. NodeInfos can be found via the network map
+ * cache, accessible from a [net.corda.core.node.services.NetworkMapCache].
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/node/NodeInfo.kt]: They're also available from RPC `networkMapSnapshot()`

File: client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt (hunk 1 above)
Comment (mikehearn) [reply client/rpc/src/main/kotlin/net/corda/client/rpc/CordaRPCClient.kt]: Visible in the IDE? I know they're visible at the bytecode level via reflection.

File: .idea/scopes/Corda_API.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<component name="DependencyValidationManager">
+  <scope name="Corda API" pattern="src[core_main]:*..*&amp;&amp;!src[core_main]:net.corda.core.internal..*||src[rpc_main]:net.corda.client.rpc.*||src[jackson_main]:net.corda.client.jackson.*" />
```
[the remark this answers is not in the document]
Comment (mikehearn) [reply .idea/scopes/Corda_API.xml]: Nope. The Scope language is a bit odd. This is correct. Once merged, open up your preferences and look at Appearance & Behaviour > Scopes. You will see that rpc.internal is black (not included). The reason is you need two dots to trigger recursive capture.

**Pull request merged**